cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -338,11 +342,28 @@ runSetupMethod SelfExecMethod = selfExecSetupMethod
 runSetup :: Verbosity -> Setup
          -> [String]  -- ^ command-line arguments
          -> IO ()
-runSetup verbosity setup args =
-  let method = setupMethod setup
-      options = setupScriptOptions setup
-      bt = setupBuildType setup
-  in runSetupMethod method verbosity options bt args
+runSetup verbosity setup args = do
+    runSetupMethod method verbosity options bt args'
+  where
+    method = setupMethod setup
+    options = setupScriptOptions setup
+    bt = setupBuildType setup
+
+    -- Hack to rewrite new-style --verbose flags
+    -- if they happen to reach this point for pre-Cabal-1.25
+    --
+    -- TODO: figure out a less hacky-way to accomplish this without
+    -- adding "when CabalLibVersion >= 1.25 then dropVerbosityFlags"
+    -- checks all over the place...
+    args' | setupVersion setup >= mkVersion [1,25] = args
+          | otherwise = map fixupVerboseFlag args
+
+    fixupVerboseFlag :: String -> String
+    fixupVerboseFlag s0
```

Comment:
As mentioned on IRC, it'd be better to provide a helper function here that the callers of setupWrapper can use to adjust their verbosity flag. This parsing of strings is ugly.